multiplexer/abci/multiplexer.go

```diff
@@ -0,0 +1,662 @@
+	return m.nativeApp != nil
+}
+
+// isEmbeddedApp checks if an embedded application is currently running.
+func (m *Multiplexer) isEmbeddedApp() bool {
+	return !m.isNativeApp()
+}
+
+// isGrpcOnly checks if the GRPC-only mode is enabled using the configuration flag.
+func (m *Multiplexer) isGrpcOnly() bool {
+	return m.svrCtx.Viper.GetBool(flagGRPCOnly)
+}
+
+// registerCleanupFn enables the registration of additional cleanup functions that get called during Cleanup
+func (m *Multiplexer) registerCleanupFn(cleanUpFn func() error) {
+	m.cleanupFns = append(m.cleanupFns, cleanUpFn)
+}
+
+func (m *Multiplexer) Start() error {
+	m.g, m.ctx = getCtx(m.svrCtx, true)
+
+	emitServerInfoMetrics()
+
+	// startApp starts the underlying application, either native or embedded.
+	if err := m.startApp(); err != nil {
+		return err
+	}
+
+	if m.isGrpcOnly() {
+		m.logger.Info("starting node in gRPC only mode; CometBFT is disabled")
+		m.svrCfg.GRPC.Enable = true
+	} else {
+		m.logger.Info("starting comet node")
+		if err := m.startCmtNode(); err != nil {
+			return err
+		}
+	}
+
+	if m.isEmbeddedApp() {
+		m.logger.Debug("using embedded app, not continuing with grpc or api servers")
+		return m.g.Wait()
+	}
+
+	if err := m.enableGRPCAndAPIServers(m.nativeApp); err != nil {
+		return err
+	}
+
+	// wait for signal capture and gracefully return
+	// we are guaranteed to be waiting for the "ListenForQuitSignals" goroutine.
+	return m.g.Wait()
+}
+
+// enableGRPCAndAPIServers enables the gRPC and API servers for the provided application if configured to do so.
+// It registers transaction, Tendermint, and node services, and starts the gRPC and API servers if enabled.
+func (m *Multiplexer) enableGRPCAndAPIServers(app servertypes.Application) error {
+	if app == nil {
+		return fmt.Errorf("unable to enable grpc and api servers, app is nil")
+	}
+	// if we are running natively and have specified to enable gRPC or API servers
+	// we need to register the relevant services.
+	if m.svrCfg.API.Enable || m.svrCfg.GRPC.Enable {
+		m.logger.Debug("registering services and local comet client")
+		m.clientContext = m.clientContext.WithClient(local.New(m.cmNode))
+		app.RegisterTxService(m.clientContext)
+		app.RegisterTendermintService(m.clientContext)
+		app.RegisterNodeService(m.clientContext, m.svrCfg)
+	}
+
+	// startGRPCServer the grpc server in the case of a native app. If using an embedded app
+	// it will use that instead.
+	if m.svrCfg.GRPC.Enable {
+		grpcServer, clientContext, err := m.startGRPCServer()
+		if err != nil {
+			return err
+		}
+		m.clientContext = clientContext // update client context with grpc
+
+		// startAPIServer starts the api server for a native app. If using an embedded app
+		// it will use that instead.
+		if m.svrCfg.API.Enable {
+			metrics, err := startTelemetry(m.svrCfg)
+			if err != nil {
+				return err
+			}
+
+			if err := m.startAPIServer(grpcServer, metrics); err != nil {
+				return err
+			}
+		}
```

Comment:
[question] why does this only happen inside the `if m.svrCfg.GRPC.Enable {` block? Is it possible to enable API and not GRPC? If not, should we error early if a user passes the CLI flags: `--api.enable true --grpc.enable false`?